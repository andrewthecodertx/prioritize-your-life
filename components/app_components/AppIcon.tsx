import React from "react";
import Ionicons from "@expo/vector-icons/Ionicons";
import { type IconProps } from "@expo/vector-icons/build/createIconSet";
import { type ComponentProps } from "react";
import { View } from "react-native";
import { ScaledSheet, s } from "react-native-size-matters";


/**
 * Renders an application icon.
 *
 * @component
 * @param {object} props - The component props.
 * @param {object} props.style - The style object to apply to the container view.
 * @param {string} props.rest - The rest of the props to pass to the Ionicons component.
 * @returns {JSX.Element} The rendered AppIcon component.
 */
export function AppIcon({
  style,
  ...rest
}: IconProps<ComponentProps<typeof Ionicons>["name"] | any>) {
  return (
    <View style={[styles.container, style]}>
      <Ionicons size={s(28)} {...rest} />
    </View>
  );
}

const styles = ScaledSheet.create({
  container: {
    alignItems: "center",
    backgroundColor: "transparent",
    borderRadius: 100,
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    padding: s(2.5),
  },
});
