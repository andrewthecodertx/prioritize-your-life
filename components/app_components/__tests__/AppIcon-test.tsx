import * as React from "react";
import renderer from "react-test-renderer";
import { AppIcon } from "../AppIcon";

describe("AppIcon Tests", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it(`renders correctly`, () => {
    const tree = renderer
      .create(<AppIcon name="home" size={30} color="red" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
