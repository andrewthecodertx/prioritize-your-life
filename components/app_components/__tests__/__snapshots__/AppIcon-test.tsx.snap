// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIcon Tests renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 100,
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 5.357142857142857,
      },
      undefined,
    ]
  }
>
  <Text />
</View>
`;
